### DISCUSSION TITLE
Rigidbodys doesn´t stop sleeping when the body they are leaning on moves

### DISCUSSION THREAD

**Discussion by CebGIN** [godot-jolt/godot-jolt#933] [Questions] [answered]
When a Rigidbody is leaning on my player and it remains still, the rigidbody starts sleeping, but when the players moves, the rigidbody keeps floating, I´m using jolt 0.13.0 beta 1 and Godot 4.3 rc3, someone knows how I can fix it?

**Comment by mihe** [comment, marked as answer]
This is something you'll see with Godot Physics as well. I believe this is largely meant to be a feature, as it prevents bodies from waking up and costing performance when streaming stuff in/out.

The only real way to fix it would be to do a shape query around the player (a `ShapeCast3D` node with zero motion works as well) and then explicitly set the `sleeping` property of any intersecting `RigidBody3D` to `false`.

**Reply by CebGIN** [reply]
What I did at that time was coneccting to the on body leave signal an fuction that would wake up the RB, althoug, I'm now working In a Physics based character controller, that seems to not have this problem
